Client/src/hooks/useActivity.js

```diff
@@ -0,0 +1,124 @@
+import { useEffect, useState } from "react";
+import {
+  getUserActivities,
+  addActivity,
+  updateActivity,
+  deleteActivity,
+} from "../services/ActivityService.js";
+import { useAuth } from "../contexts/AuthContext.jsx";
+import { useNavigate } from "react-router-dom";
+
+const useActivity = (userId) => {
```

Comment:
useActivity should take activityId not user Id and it should be optional (so that we can use useActivity to create and activity)